modules/formulize/class/durationElement.php

```diff
@@ -0,0 +1,271 @@
+<?php
+require_once XOOPS_ROOT_PATH . "/modules/formulize/class/elements.php";
+require_once XOOPS_ROOT_PATH . "/modules/formulize/include/functions.php";
+
+class formulizeDurationElement extends formulizeElement
+{
+
+	function __construct()
+	{
+		$this->name = "Duration";
+		$this->hasData = true;
+		$this->needsDataType = false; // We'll force this to be stored as INT (minutes)
+		$this->overrideDataType = "int"; // Store as integer minutes
+		$this->adminCanMakeRequired = true;
+		$this->alwaysValidateInputs = true; // We'll always validate to ensure duration is within bounds
+		$this->canHaveMultipleValues = false;
+		$this->hasMultipleOptions = false;
+		parent::__construct();
+	}
+}
+
+class formulizeDurationElementHandler extends formulizeElementsHandler
+{
+
+	var $db;
+
+	function __construct($db)
+	{
+		$this->db = &$db;
+	}
+
+	// Conversion factors to minutes
+	private $timeUnits = array(
+		'days' => 1440,
+		'hours' => 60,
+		'minutes' => 1
+	);
+
+	function create()
+	{
+		return new formulizeDurationElement();
+	}
+
+	// Prepare admin UI data
+	function adminPrepare($element)
+	{
+		$dataToSendToTemplate = array();
+
+		if (is_object($element) && is_subclass_of($element, 'formulizeelement')) {
```

Comment:
I think some things (like class names?) are actually not case sensitive in PHP? But stylistically, I believe we are using formulizeElement everywhere else?